feat(mcl): Port `host-info` to maOS

PetarKirov opened the pull request:
- **refactor(mcl): Move all source one level up**
- **feat(mcl.commands.host_info): Initial macOS port**

PetarKirov committed:
feat(mcl.commands.host_info): Initial macOS port

reo101 commented on packages/mcl/src/mcl/commands/host_info.d:
```diff
@@ -649,54 +791,97 @@ struct DisplayInfo
+                        d.modes ~= disp["_spdisplays_pixels"].str;
                     }
+                    // Extract refresh rate from resolution string (e.g., "1728 x 1117 @ 120.00Hz")
+                    if (d.resolution.indexOf("@") != -1)
+                    {
+                        auto parts = d.resolution.split("@");
+                        if (parts.length > 1)
+                            d.refreshRate = parts[1].strip;
+                    }
```
https://dlang.org/phobos/std_format_read.html#.formattedRead

reo101 replied:
Will do later

PetarKirov committed:
ci(gh-actions/reusable-flake-checks-ci-matrix): Fix `build` job condition

Use `always()` to ensure that `build` is executed even when `eval-matrix` and `merge-matrices` are skipped.

PetarKirov committed:
ci(gh-actions/reusable-flake-checks-ci-matrix): Fix `build` job's runs-on
